src/pudl/metadata/resources/ferc1_eia_record_linkage.py

```diff
@@ -127,33 +127,34 @@
         "etl_group": "outputs",
         "field_namespace": "eia",
     },
-    # "out_eia__yearly_plant_parts_plant_gen_assn": {
-    #     "description": """In order to easily determine what generator records are associated with every
-    # plant part record, we made this association table. This table associates every plant part
-    # record (identified as ``record_id_eia``) to the possibly many 'plant_gen' records
-    # (identified as ``record_id_eia_plant_gen``).""",
-    #     "schema": {
-    #         "fields": [
-    #             "record_id_eia",
-    #             "record_id_eia_plant_gen",
-    #             "report_date_plant_gen",
-    #             "operational_status_pudl_plant_gen",
-    #             "energy_source_code_1_plant_gen",
-    #             "ferc1_generator_agg_id_plant_gen",
-    #             "prime_mover_code_plant_gen",
-    #             "unit_id_pudl_plant_gen",
-    #             "technology_description_plant_gen",
-    #             "ferc_acct_name_plant_gen",
-    #             "plant_id_eia_plant_gen",
-    #             "generator_operating_year_plant_gen",
-    #             "generator_id_plant_gen",
-    #         ],
-    #         "primary_key": ["record_id_eia", "record_id_eia_plant_gen"],
-    #         "sources": ["eia860", "eia923"],
-    #         "etl_group": "outputs",
-    #         "field_namespace": "eia",
-    #     },
-    # },
+    "out_eia__yearly_plant_parts_plant_gen_assn": {
```

Comment:
Should this be an `out_eia__` or an `out_pudl__` table?  The FERC-EIA plant parts association is currently `out_pudl__`

For naming consistency I think we want to move the `assn` to the beginning after the double underscore as an indication of what kind of table it is. Though sadly when there's a time frequency we haven't be consistent about whether it comes first or second so there are both `__assn_yearly_` and `__yearly_assn_` tables:

```
core_eia860__assn_boiler_cooling
core_eia860__assn_boiler_generator
core_eia860__assn_boiler_stack_flue
core_eia860__assn_yearly_boiler_emissions_control_equipment
core_eia861__assn_balancing_authority
core_eia861__assn_utility
core_epa__assn_eia_epacamd
core_epa__assn_eia_epacamd_subplant_ids
core_pudl__assn_eia_pudl_plants
core_pudl__assn_eia_pudl_utilities
core_pudl__assn_ferc1_dbf_pudl_utilities
core_pudl__assn_ferc1_pudl_plants
core_pudl__assn_ferc1_pudl_utilities
core_pudl__assn_ferc1_xbrl_pudl_utilities
core_pudl__assn_utilities_plants
out_pudl__yearly_assn_eia_ferc1_plant_parts
```